docs/operators/fsInputOperator.md

```diff
@@ -0,0 +1,101 @@
+| *scanner* | Scanner to scan new files in directory | [DirectoryScanner](#DirectoryScanner) | No | DirectoryScanner |
+
+#### <a name="JDBCStore"></a>Properties of DirectoryScanner
+| **Property** | **Description** | **Type** | **Mandatory** | **Default Value** |
+| -------- | ----------- | ---- | ------------------ | ------------- |
+| *filePatternRegexp* | regex to select files from input directory | String | No | N/A |
+
+## Abstract Methods
+void emit(T tuple): Abstract method that emits tuple read from file.
+T readEntity(): Abstract method to read file entity (can be a line or block).
+
+## Partitioning of File Input Operator
+Configure parameter "partitionCount" to define desired number of partitions.
+
+```xml
+  <property>
+    <name>dt.operator.{OperatorName}.prop.partitionCount</name>
+    <value>4</value>
+  </property>
+```
+
+where {OperatorName} is the name of the input operator.
+Above lines will partition operator statically 4 times. Above value can be changed accordingly to change the number of static partitions.
+
+#### Dynamic Partitioning of JDBC Poller
+Implement ```getNewPartitionCount()``` method to do dymanic partitioning.
+
+## Derived Classes
+### 1. AbstractFTPInputOperator
+The class is used to read files from FTP file system.
+#### Properties, Attributes and Ports
+#### <a name="props"></a>Properties of AbstractFTPInputOperator
+| **Property** | **Description** | **Type** | **Mandatory** | **Default Value** |
+| -------- | ----------- | ---- | ------------------ | ------------- |
+| *host*| Hostname of ftp server.| String | Yes | N/A |
+| *port*| Port of ftp server.| int | No | 21 (default ftp port) |
+| *userName*| Username which is used for login to the server. | String | No | anonymous |
+| *password*| Password which is used for login to the server. | String | No | gues |
+
+### 2. AbstractParquetFileReader
+Reads Parquet files from input directory using GroupReadSupport. Derived classes need to implement [convertGroup(Group)](https://www.datatorrent.com/docs/apidocs/com/datatorrent/contrib/parquet/AbstractParquetFileReader.html#convertGroup(Group)) method to convert Group to other type.
+
+#### Properties, Attributes and Ports
+#### <a name="props"></a>Properties of AbstractParquetFileReader
+| **Property** | **Description** | **Type** | **Mandatory** | **Default Value** |
+| -------- | ----------- | ---- | ------------------ | ------------- |
+| *parquetSchema*| Parquet Schema to parse record. | String | Yes | N/A |
+
+### 3. AbstractThroughputFileInputOperator
+This operator provides dynamic partitioning to AbstractFileInputOperator. The user can set the preferred number of pending files per operator as well as the max number of operators and define a repartition interval. If a physical operator runs out of files to process and an amount of time greater than or equal to the repartition interval has passed then a new number of operators are created to accommodate the remaining pending files.
```

Comment:
This is not clear and needs to be described in greater detail. Where and how does throughput enter the picture ? Perhaps add a couple of specific scenarios and describe what happens there.